### ISSUE TITLE
RFC: Transparent remote (ssh) access similar to mc-fish, emacs-tramp, or vim-netrw

### ISSUE THREAD

**Issue by kiryph** [vifm/vifm#317]
I have added following configuration to my `.vifm/vifmrc`:
```viml
" Add simple support for ssh remote
" :ssh user@remote:/path/to/remote/file
filetype *.ssh FUSE_MOUNT2|sshfs %PARAM %DESTINATION_DIR
command ssh
    \ : let $SSH_FILE = system('mktemp /tmp/XXXX.ssh')
    \ | execute '!echo ''%a'' > $SSH_FILE; ls $SSH_FILE %%U'
    \ | execute 'normal l'
    \ | execute '!rm $SSH_FILE'
```
This allows me to connect to a remote server with following command:
```
:ssh user@remote:/path/to/remote/file
```
However, this solution is unsatisfying. For example, I cannot create a bookmark or use the directory history to jump back to this folder with `Ctrl-o/i`.

I am aware of following projects which try to provide transparent remote access:

- tramp in *emacs* (https://www.gnu.org/software/tramp/#Overview)
- fish in *mc/midnight-commander* (https://en.wikipedia.org/wiki/Files_transferred_over_shell_protocol)
- and the plugin *netrw* in vim (vim :help netrw-transparent)

Are there any plans for vifm to enhance the user experience when accessing remote files?

**Comment by xaizek** [comment]
> Are there any plans for vifm to enhance the user experience when accessing remote directories?

Not really. Currently FUSE and regular paths are treated mostly uniformly and handling FUSE paths in a special way would require reworking things to at least have some addressing system for non-regular paths. Except for opening/closing things should be quite transparent actually (well, path prefix and the fact that it might change can be somewhat annoying).

> For example, I cannot create a bookmark or use the directory history to jump back to this folder with `Ctrl-o/i`.

That's because you closed it by leaving the directory explicitly, maybe this what needs to be addressed. If you won't go to parent directory from root of a FUSE mount, it will remain active and everything else should work as usual.

**Comment by kiryph** [comment]
Thanks for your response.

I have one thought: would it be possible to create a deterministic temp folder for sshfs keeping the .ssh file which can be reused when returning to this folder.

Does this sound any good?

**Comment by xaizek** [comment]
Deterministic name would probably be awfully long, storing meta-data that could map directory names to mount options might be another way of doing this. Files themselves aren't required for mounts as long as there is another source of parameters. However, there still might be conflicts between multiple instances of vifm using the same mount.
